src/seedu/addressbook/Main.java

```diff
@@ -106,11 +106,14 @@ private void recordResult(CommandResult result) {
      * @return result of the command
      */
     private CommandResult executeCommand(Command command)  {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
+            return result;
         } catch (Exception e) {
```

Comment:
As discussed in tutorial, it would be better if all exceptions are handled individually.